src/box/lua/config/cluster_config.lua

```diff
@@ -171,6 +172,12 @@ local name = schema.scalar({
     end,
 })
 
+local function name_with_description(field_path)
+    local schema_obj = name
+    schema_obj.description = annotations[field_path]
+    return schema_obj
+end
```

Comment:
The `local schema_obj = name` assignment doesn't perform a copying. Looks suspectful. I guess that the description is finally the same for all the `name_with_description()` return values.

Reply:
Fixed.

Reply:
Hmm. The trick with `__index` is interesting. To be honest, I would not use it here. It is different from copying in how the serialization works, how `pairs()` works and strictly speaking it voilates the schema node format. And we don't need to do optimizations here: this code is called four times.

Also I think that we don't need to add descriptions to map keys, see [here](https://github.com/tarantool/tarantool/pull/10629#discussion_r1794184101].

Reply:
Irrelevant code (removed).